port/zos390/omrvmem.c

```diff
@@ -500,26 +500,28 @@ reserve1MPageableBelowBar(struct OMRPortLibrary *portLibrary, J9PortVmemIdentifi
  *
  * @params[in] portLibrary the portLibrary
  * @params[out] identifier the vmem identifier to be updated with the results of the allocation attempt
- * @params[in] mode the mode
- * @params[in] byteAmount the size to allocate
- * @params[in] pageSize the pageSize to request
- * @params[in] pageFlags flags indicating type of page to be used
- * @params[in] options flags to indicate allocation strategy
+ * @params[in] params contains the memory reservation parameters including mode, byteAmount, pageSize, pageFlags, options, and etc.
  * @params[in] category Memory category
  *
  * @return NULL on failure, otherwise pointer to the memory that was allocated.
  */
 static void *
-reservePagesAboveBar(struct OMRPortLibrary *portLibrary, J9PortVmemIdentifier *identifier, uintptr_t mode, uintptr_t byteAmount, uintptr_t pageSize, uintptr_t pageFlags, uintptr_t options, OMRMemCategory *category)
+reservePagesAboveBar(struct OMRPortLibrary *portLibrary, J9PortVmemIdentifier *identifier, struct J9PortVmemParams *params, OMRMemCategory *category)
 {
 	/* The maximum amount of memory that can be allocated with userExtendedPrivateAreaMemoryType detected for 2To32G and 2To64 */
 #define MAX_2TO32G_MEMORY_IN_MB ((UDATA)30*1024)	/* unit is MB i.e. 30*1024 MB = 30 GB */
 #define MAX_2TO64G_MEMORY_IN_MB ((UDATA)62*1024)	/* unit is MB i.e. 62*1024 MB = 62 GB */
 
+	uintptr_t mode       = params->mode;
+	uintptr_t byteAmount = params->byteAmount;
+	uintptr_t pageSize   = params->pageSize;
+	uintptr_t pageFlags  = params->pageFlags;
+	uintptr_t options    = params->options;
+	void* startAddress   = (void*)params->startAddress;
```

Comment:
We should add assertion in `omrvmem_reserve_memory_ex()` to check that `startAddress` is aligned to `params->pageSize`:
```
	ASSERT_VALUE_IS_PAGE_SIZE_ALIGNED(params.byteAmount, params.pageSize); <-- exists
	ASSERT_VALUE_IS_PAGE_SIZE_ALIGNED(params.startAddress, params.pageSize); <-- new 
```

Reply:
**startAddress** alignment check added in b333ae71e85896febf6ce8260d2d28493524dddf